WhateverGreen/kern_igfx.hpp

```diff
@@ -457,7 +492,668 @@ class IGFX {
 	/**
 	 *  ReadAUX wrapper to modify the maximum link rate value in the DPCD buffer
 	 */
-	static int wrapReadAUX(void *that, IORegistryEntry *framebuffer, uint32_t address, uint16_t length, void *buffer, void *displayPath);
+	static IOReturn wrapReadAUX(void *that, IORegistryEntry *framebuffer, uint32_t address, uint16_t length, void *buffer, void *displayPath);
+	
+	/**
+	 *	Reflect the `AppleIntelFramebufferController::CRTCParams` struct
+	 *
+	 *	@note Unlike the Intel Linux Graphics Driver,
+	 *	- Apple does not transform the `pdiv`, `qdiv` and `kdiv` fields.
+	 *	- Apple records the final central frequency divided by 15625.
+	 *	@ref static void skl_wrpll_params_populate(params:afe_clock:central_freq:p0:p1:p2:)
+	 *	@seealso https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/gpu/drm/i915/intel_dpll_mgr.c?h=v5.1.13#n1171
+	 */
+	struct CRTCParams
```

Comment:
We probably need to pack this struct or use a static_assert to guarantee the correct size when passed to driver (even though it's unlikely to break on X86)

Reply:
We definitely do not want to pack this, but just as @07151129 static assertions are needed to verify offsets.
Please add code like `static_assert(offsetof(CRTCParams, pdiv) == 0x20), "Invalid pdiv offset, check your compiler");` for critical areas.